src/components/DatasetCounter.tsx

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+"use client";
```

Comment:
I dont think it has to be a client side component. It can be rendered on server side

Reply:
I think it has to be client component due to useState and useEffect

Reply:
Works rendered on the server side.